src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -188,6 +191,8 @@ public void Dispose()
 
         internal object CaptureDisposable(object service)
         {
+            _captureDisposableCallback?.Invoke(service);
```

Comment:
That's not enough. It's about calls to this method not captured disposables. In the cases above, the transient services aren't disposable.